destination_middleware.go

```diff
@@ -58,6 +76,47 @@ const (
 
 type ctxKeyBatchEnabled struct{}
 
+// DestinationWithBatchConfig is the configuration for the
+// DestinationWithBatch middleware. Fields set to their zero value are
+// ignored and will be set to the default value.
+type DestinationWithBatchConfig struct {
+	// BatchSize is the default value for the batch size.
+	BatchSize int
+	// BatchDelay is the default value for the batch delay.
+	BatchDelay time.Duration
+}
```

Comment:
WDYT about setting the default values through Wrap(), as is done in the source middleware?

Reply:
Setting the default values through `Wrap` is ok with me. The question is _if_ we should set the defaults for batches, and if we do, what would be sane defaults? 😉 I think this might be a conversation for a discussion, as the implementation is trivial, we just need to consider what impact this could have on pipelines and what are sane defaults.

Reply:
I was assuming that we are somewhere along way setting the batch size to 1. But if it works in a different, as you said,  no need to go into details here. :+1:

Reply:
I actually went and refactored this middleware a bit, so that it makes more sense and is more aligned with the other middleware, but didn't change the behavior or defaults. The zero values stayed the default, batching only kicks in if you set the batch size to something bigger than 1 or the delay to something bigger than 0.